etcdserver/apply.go

```diff
@@ -336,13 +350,17 @@ func (a *applierV3backend) Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error) {
 // applyCompare applies the compare request.
 // It returns the revision at which the comparison happens. If the comparison
 // succeeds, the it returns true. Otherwise it returns false.
-func (a *applierV3backend) applyCompare(c *pb.Compare) (int64, bool) {
+func (a *applierV3backend) applyCompare(header *pb.RequestHeader, c *pb.Compare) (int64, bool, error) {
```

Comment:
combine the bool and error? instead of return false we can return `errCompareFailed`